compiler/gcc-cross-compile.cmake

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2015 Alexandre Pretyman
+# All rights reserved.
+#
+# ------------------------------------------------------------------------------
+#  GCC Based Cross Compiler CMake toolchain file
+#
+#  Variables:
+#   CROSS_COMPILE_TOOLCHAIN_PATH=/path/to/toolchain/bin
+#   CROSS_COMPILE_TOOLCHAIN_PREFIX=arm-unknown-linux-gnueabihf
+#   CROSS_COMPILE_SYSROOT=/path/to/sysroot [optional]
+# ------------------------------------------------------------------------------
+
+if(DEFINED POLLY_COMPILER_GCC_CROSS_COMPILE)
+  return()
+else()
+  set(POLLY_COMPILER_GCC_CROSS_COMPILE TRUE)
+endif()
+
+if( CMAKE_TOOLCHAIN_FILE )
+  # touch toolchain variable to suppress "unused variable" warning
+endif()
+
+string(COMPARE EQUAL
+    "${CROSS_COMPILE_TOOLCHAIN_PATH}"
+    ""
+    _is_empty
+)
+if(_is_empty)
+  polly_fatal_error("CROSS_COMPILE_TOOLCHAIN_PATH not set. Set it to the absolute path of the \"bin\" directory for the toolchain")
+endif()
+
+string(COMPARE EQUAL
+    "${CROSS_COMPILE_TOOLCHAIN_PREFIX}"
+    ""
+    _is_empty
+)
+if(_is_empty)
+  polly_fatal_error("CROSS_COMPILE_TOOLCHAIN_PREFIX not set. Set it to the triplet of the toolchain")
+endif()
+
+
+#Check for if sysroot exists, optional, since it could be hardcoded in the compiler
+string(COMPARE NOTEQUAL "${CROSS_COMPILE_SYSROOT}" "" _has_sysroot)
```

Comment:
hum... well, strictly speaking it is not fully required, but it is a very good idea to have it set, as when compiling Qt for the Raspberry Pi, it seems to be needed. For now I guess you are right and it is best to make it required, because having it required would make things just work, leaving it optional could make the toolchain not work.